dockstore-webservice/scripts/sortOpenAPIYaml.sh

```diff
@@ -3,4 +3,10 @@ set -o errexit
 set -o pipefail
 set -o nounset
 # https://github.com/mikefarah/yq Require version 3.1.2.  Newer version will likely also work.
-yq r --tojson src/main/resources/openapi3/unsortedopenapi.yaml | yq r - -P > src/main/resources/openapi3/openapi.yaml
+yq r --tojson src/main/resources/openapi3/unsortedopenapi.yaml | yq r - -P > src/main/resources/openapi3/openapi.yaml || {
```

Comment:
Seems odd/redundant that this needs to be done in script. 
It is more "Maveny" to configure this in the Maven exec plugin and the plugin should fail if this script exits non-zero. 
https://www.mojohaus.org/exec-maven-plugin/exec-mojo.html#successCodes

Reply:
doesn't it already do that?

```
[INFO] --- exec-maven-plugin:1.6.0:exec (Sort OpenAPI.yaml) @ dockstore-webservice ---
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/commons/commons-exec/1.3/commons-exec-1.3.jar
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/commons/commons-exec/1.3/commons-exec-1.3.jar (0 B at 0 B/s)
/dockstore-webservice/scripts/sortOpenAPIYaml.sh: line 6: yq: command not found
/dockstore-webservice/scripts/sortOpenAPIYaml.sh: line 6: yq: command not found
[ERROR] Command execution failed.
org.apache.commons.exec.ExecuteException: Process exited with an error: 127 (Exit value: 127)
    at org.apache.commons.exec.DefaultExecutor.executeInternal (DefaultExecutor.java:404)
    at org.apache.commons.exec.DefaultExecutor.execute (DefaultExecutor.java:166)
    at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:804)
    at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:751)
    at org.codehaus.mojo.exec.ExecMojo.execute (ExecMojo.java:313)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for dockstore 1.9.0-beta.1-SNAPSHOT:
[INFO]
[INFO] dockstore .......................................... SUCCESS [10:50 min]
[INFO] dockstore-common ................................... SUCCESS [ 43.637 s]
[INFO] dockstore-language-plugin-parent ................... SUCCESS [  5.659 s]
[INFO] swagger-java-quay-client ........................... SUCCESS [ 25.420 s]
[INFO] swagger-java-sam-client ............................ SUCCESS [ 11.226 s]
[INFO] swagger-java-bitbucket-client ...................... SUCCESS [ 22.629 s]
[INFO] swagger-java-discourse-client ...................... SUCCESS [ 14.118 s]
[INFO] swagger-java-zenodo-client ......................... SUCCESS [  9.545 s]
[INFO] dockstore-webservice ............................... FAILURE [01:08 min]
[INFO] swagger-java-client ................................ SKIPPED
[INFO] openapi-java-client ................................ SKIPPED
[INFO] dockstore-integration-testing ...................... SKIPPED
[INFO] dockstore-event-consumer ........................... SKIPPED
[INFO] reports ............................................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  14:32 min
[INFO] Finished at: 2020-06-23T18:25:48Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.6.0:exec (Sort OpenAPI.yaml) on project dockstore-webservice: Command execution failed.: Process exited with an error: 127 (Exit value: 127) -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :dockstore-webservice
The command '/bin/sh -c mvn clean install -DskipTests' returned a non-zero code: 1
```

Reply:
I was under the impression no, it seemed like Maven got past it in https://oicr.slack.com/archives/C16ET3CF4/p1591799974401000 but maybe there is something special about that environment?

Reply:
this is from the Dockerfile at root btw (with the yq part commented out)

Reply:
@denis-yuen So that error snippet I shared in that slack link is actually the result of me attempting to build without `yq` twice in a row. A side-effect of `yq` is wiping the contents of `openapi.yaml` (see line 6 above), so that resulted in the different error log.

The crux of this issue was a combination of me being new to maven & incorrect assumptions about the build process. The intent of this new build message is to make it explicitly clear that `yq` missing wasn't the build's fault.

Also the reason why I made this a Bash code change rather than a "Maveny" change was that I couldn't see any other way to include such a message via https://www.mojohaus.org/exec-maven-plugin/exec-mojo.html which runs the script.